frontend/src/app/app.component.ts

```diff
@@ -131,5 +91,28 @@ export class AppComponent implements OnInit {
         return hour < 12 ? 'AM' : 'PM';
       }
     });
-   }
+  }
+
+  getBrowserName() {
+    const  userAgent = navigator.userAgent;
+    let browserInfo = userAgent.match (/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || [];
+    let parsedBrowserInfo;
+
+        if (/trident/i.test(browserInfo[1])) {
+          parsedBrowserInfo = /\brv[ :]+(\d+)/g.exec(userAgent) || [];
+          return { name: 'IE', version: (parsedBrowserInfo[1] || '') };
+        }
+        if (browserInfo[1] === 'Chrome') {
+        parsedBrowserInfo = userAgent.match (/\bOPR|Edge\/(\d+)/);
+        if (parsedBrowserInfo != null)   {
+            return { name: 'Opera', version: parsedBrowserInfo[1] };
+          }
+        }
+      browserInfo = browserInfo[2] ? [browserInfo[1], browserInfo[2]] : [navigator.appName, navigator.appVersion, '-?'];
+      if (( parsedBrowserInfo = userAgent.match(/version\/(\d+)/i)) != null) {
+        browserInfo.splice(1, 1, parsedBrowserInfo[1]);
+    }
```

Comment:
Please fix the code style here to be consistent with the rest of the codebase. The indentation is off on a few lines.